### PR TITLE
Header filtering config

### PR DESCRIPTION
**Pull request by aklish**
## Description
Adds configuration to enable/disable security credential stripping in HTTP headers.

## Motivation and Context
Elide automatically removes Authorization headers from client requests before the framework is invoked.  Some users want to leverage the credentials in these headers to invoke other services from life cycle hooks or data stores.

## How Has This Been Tested?
New IT tests.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+        ArgumentCaptor<MultivaluedMap<String, String>> requestParamsCaptor = ArgumentCaptor.forClass(MultivaluedMap.class);
+        ArgumentCaptor<Map<String, List<String>>> requestHeadersCleanedCaptor = ArgumentCaptor.forClass(Map.class);
+        verify(elide.getElide()).post(any(), any(), any(), requestParamsCaptor.capture(), requestHeadersCleanedCaptor.capture(), any(), any(), any());
+
+        MultivaluedHashMap<String, String> expectedRequestParams = new MultivaluedHashMap<>();
+        expectedRequestParams.put(SORT_PARAM, ImmutableList.of("name", "description"));
+        assertEquals(expectedRequestParams, requestParamsCaptor.getValue());
+
+        assertTrue(requestHeadersCleanedCaptor.getValue().containsKey("authorization"));
```
Comment (moizarafat) [inline elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java:117]: shouldn't the cleaned headers not contain authorization and proxy-authorization? Maybe the variable is confusing..

Comment (aklish) [reply elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java:117]: This is testing identity - where we disable header cleansing.

Comment (aklish) [reply elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java:117]: There is another IT test where we test cleansing.

Comment (moizarafat) [reply elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderIdentityTest.java:117]: ahh okie.. sounds good.

**Review by moizarafat — APPROVED** [review]

**Pull request merged**